c10/util/string_view.h

```diff
@@ -591,6 +591,21 @@ constexpr inline void swap(
 
 using string_view = basic_string_view<char>;
 
+// NOTE: In C++20, this function should be replaced by str.starts_with
+constexpr bool string_view_starts_with(
+    std::string_view str,
+    std::string_view prefix) noexcept {
+  return str.find(prefix) == 0;
```

Comment:
```suggestion
  return str.rfind(prefix, 0) == 0
```
your current code could find the prefix at any point in the substring, not just at index 0, which is not the desired behavior.